feat(wifaqul ulama): support wifaqul ulama uk prayer times

hsravat-4590 opened the pull request:
This Pull Request adds support for [Wifaqul Ulama UK's calculation criteria](https://www.wifaqululama.co.uk/salahtimes)

It notably includes the following key changes:

* New United Kingdom Calculation Method 
* Support for Aqrab Yaum for High Latitude Fajr Calculations during periods of perpetual twilight

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): adds aqrabul ayyam high latitude fajr param

Add new HighLatitudeFajrRule for Aqrabul Ayyam during High Latitude Perpetual Twilight.

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): adds wifaqul ulama params

Adds a new Calculation Method used by Wifaqul Ulama UK

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): allow specification of fajr angle for calculation

Add specification of Fajr Angle for Aqrabul Youm Calculation

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): fix dhuhr adjustment value

Dhuhr Adjustment should be 4 minutes.

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update docs

Update METHODS.md

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update tests

Add some Unit Tests

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): add guard

Don't invoke Aqrabul Youm resolver unless in perpetual twilight

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): fix failures

Fix failing tests

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): fix failures

Fix failing tests

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): optimise time components

Optimise TimeComponents.utcDate function to reduce line count in PrayerTimes.ts

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): optimise time components

Removes redundant variable

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): fix linter failures

Fix failures with Linter on Unit Tests.

z3bi commented:
Jazakallah khair for this addition to the library. I would request if you could put together a json file of prayer times published by this group to be included in our testing repo (https://github.com/batoulapps/adhan-js/tree/master/Shared/Times) so we can see if there is a difference in the times being calculated by the library and the official published times.

Please take a look at https://github.com/batoulapps/adhan-js/blob/master/Shared/Times/London-MoonsightingCommittee.json as an example. In it I have a copy of the original times and show where we had to adjust the times to account for rounding. It should be sufficient to include one day of prayer times for each month so we have a representative sample for the year.

Also, it seems that the formatting of the README was changed significantly, this is making it difficult to see what was changed. If you could revert those changes and only have the new lines added that would be much better inshAllah.

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): support independent fajr high lat - middle of the night

Wifaqul Ulama's criteria allows for Middle of the Night for Fajr but 1/7th for Isha

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update uk calculation method for correct high latitude rule

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): add comparison test data

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update unit tests for independent fajr

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update times test

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update unit tests for mid of night fajr rule

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): refactor to match naming

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): remove unused import

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): revert update docs

hsravat-4590 committed:
feat(wifaqul ulama uk fajr): update docs with correct formatting

hsravat-4590 commented:
@z3bi I've updated this PR with requested changes.

hsravat-4590 committed:
feat(wifaqul ulama): update exports

hsravat-4590 committed:
feat(wifaqul ulama):  fix issue where fajr was incorrect when aqrab youm was used

hsravat-4590 committed:
feat(wifaqul ulama):  remove console log

hsravat-4590 committed:
feat(wifaqul ulama):  add separate test for aqrab youm

hsravat-4590 committed:
feat(wifaqul ulama):  fix import

hsravat-4590 commented:
Assalamu Alaykum, any feedback on why this was closed?
